tests/robot_crawler_test.php

```diff
@@ -102,6 +102,38 @@ public function test_absolute_urls($base, $links) {
             $this->assertEquals($value, $this->robot->absolute_url($base, $key));
         }
     }
+
+    /**
+     * @return array of test cases
+     *
+     * Local and external URLs and their tricky combinations
+     */
+    public function urls_provider() {
+        global $CFG;
+        $mydomain = parse_url($CFG->wwwroot, PHP_URL_HOST);
```

Comment:
unit tests should not contain variables, test them explicitely. If you have vars there is a chance under some conditions they could pass and others fail. Make these explicit by setting $CFG->wwwroot to something provided by the provider and then running the method in the test